libc/test/src/__support/big_int_test.cpp

```diff
@@ -927,4 +928,143 @@ TEST(LlvmLibcUIntClassTest, OtherWordTypeTests) {
   ASSERT_EQ(static_cast<int>(a >> 64), 1);
 }
 
+TEST(LlvmLibcUIntClassTest, OtherWordTypeCastTests) {
```

Comment:
Can you also add tests for same bits but different word types, like between `<64, uint64_t>`, `<64, uint32_t>`, and `<64, uint16_t>`, and mixed signed, unsigned?

Reply:
Then we should have `static_assert` to prevent / detect if it happens in the future.  Otherwise, it might occur with some template parameter computations, and we don't have anything to guarantee its behavior in that case.  Just like I was surprised when seeing that `UInt<10k+>` was actually used at some point.

Reply:
I've added a static assert.